components/engine_traits/src/range_cache_engine.rs

```diff
@@ -163,6 +163,46 @@ impl CacheRegion {
     pub fn overlaps(&self, other: &CacheRegion) -> bool {
         self.start < other.end && other.start < self.end
     }
+
+    pub fn union(&self, other: &CacheRegion) -> Option<CacheRegion> {
```

Comment:
These two new methods looks weird to me under the current naming `CacheRegion`. Maybe we should split it into two struct like following:
```
struct Range {
  start: Vec<u8>,
  end: Vec<u8>,
}

struct CacheRegion {
  id: u64,
  epoch_version: u64,
  range: Range,
}
```
Thus all operations that actully take a range rather than a region should use `Range` instead, e.g, to use Range as the parameter to handle evict_by_range.

Reply:
It's like a hundred-line refactor, I'll leave it to an other PR.

Reply:
Yeah, we may handle this later, won't block this pr.